quick/commands/topic.py

```diff
@@ -97,11 +100,33 @@ def add_args(self, parser: ArgumentParser, required: ArgumentGroup, optional: Ar
             dest="immutable",
             action="store_true",
             help="An immutable topic does not allow ingesting the same key twice (default: False)",
+            required=False,
         )
         optional.add_argument(
             "--retention-time",
             type=str,
             help="Retention time of data in the topic in (if not given, the data is kept indefinitely)",
+            required=False,
```

Comment:
Is this needed? Isn't it implied by using `optional`? And why not add it to the other optional parameter you've added?

Reply:
No, it is not needed. I would also remove it because it is confusing.